api/src/DataProvider/TopBookDataProvider.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\DataProvider;
+
+use App\Entity\TopBook;
+use Symfony\Component\HttpKernel\KernelInterface;
+use Symfony\Contracts\Cache\CacheInterface;
+use Symfony\Contracts\Cache\ItemInterface;
+
+final class TopBookDataProvider
```

Comment:
It's not an API Platform DataProvider, but a class to retrieve some data. I think the best design pattern name would be Repository. But to prevent any misunderstanding with a Doctrine Repository, could we rename this class to `TopBookDataRepository`?